.github/workflows/ios_deploy_staging_to_firebase.yml

```diff
@@ -0,0 +1,55 @@
+name: ios-deploy-to-staging
+on:
+  # Trigger the workflow on push action
+  push:
+    branches:
+      - develop
+
+jobs:
+  build_and_upload_staging_app_to_firebase:
+    name: Build And Upload iOS Staging Application To Firebase
+    runs-on: macOS-latest
+    defaults:
+      run:
+        working-directory: template
+    env:
+      TEAM_ID: ${{ secrets.TEAM_ID }}
+      FASTLANE_USER: ${{ secrets.FASTLANE_USER }}
+      FASTLANE_PASSWORD: ${{ secrets.FASTLANE_PASSWORD }}
+      FASTLANE_APPLE_APPLICATION_SPECIFIC_PASSWORD: ${{ secrets.FASTLANE_APPLE_APPLICATION_SPECIFIC_PASSWORD }}
+      FASTLANE_SESSION: ${{ secrets.FASTLANE_SESSION }}
+      MATCH_PASSWORD: ${{ secrets.MATCH_PASSWORD }}
+      KEYCHAIN_PASSWORD: ${{ secrets.KEYCHAIN_PASSWORD }}
+    steps:
+    - uses: actions/checkout@v1
+
```

Comment:
@markgravity I think we need these 2 steps to `generate the actual project` before building & deploying from the root workflow, right? We don't need them in the template project for sure.

https://github.com/nimblehq/flutter-templates/blob/125e9c8609d655240cc7a0768cd4d0bf91312a94/.github/workflows/android_deploy_staging.yml#L18-L24